packages/@aws-cdk/aws-codepipeline-actions/test/integ.pipeilne-elastic-beanstalk-deploy.ts

```diff
@@ -0,0 +1,146 @@
+  managedPolicies: [
+    {
+      managedPolicyArn: 'arn:aws:iam::aws:policy/service-role/AWSElasticBeanstalkEnhancedHealth',
+    },
+    {
+      managedPolicyArn: 'arn:aws:iam::aws:policy/AWSElasticBeanstalkManagedUpdatesCustomerRolePolicy',
+    },
+  ],
+});
+
+const instanceProfileRole = new iam.Role(stack, 'instance-profile-role', {
+  roleName: 'codepipeline-elasticbeanstalk-action-test-instance-profile-role',
+  assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
+  managedPolicies: [
+    {
+      managedPolicyArn: 'arn:aws:iam::aws:policy/AWSElasticBeanstalkWebTier',
+    },
+    {
+      managedPolicyArn: 'arn:aws:iam::aws:policy/AWSElasticBeanstalkMulticontainerDocker',
+    },
+    {
+      managedPolicyArn: 'arn:aws:iam::aws:policy/AWSElasticBeanstalkWorkerTier',
+    },
+  ],
+});
+
+const instanceProfile = new iam.CfnInstanceProfile(stack, 'instance-profile', {
+  roles: [instanceProfileRole.roleName],
+  instanceProfileName: instanceProfileRole.roleName,
+});
+
+const beanstalkApp = new elasticbeanstalk.CfnApplication(stack, 'beastalk-app', {
+  applicationName: 'codepipeline-test-app',
+});
+
+const beanstalkEnv = new elasticbeanstalk.CfnEnvironment(stack, 'beanstlk-env', {
+  applicationName: beanstalkApp.applicationName!,
+  environmentName: 'codepipeline-test-env',
+  solutionStackName: '64bit Amazon Linux 2 v5.5.6 running Node.js 16',
+  optionSettings: [
+    {
+      namespace: 'aws:autoscaling:launchconfiguration',
+      optionName: 'IamInstanceProfile',
+      value: instanceProfile.instanceProfileName,
+    },
+    {
+      namespace: 'aws:elasticbeanstalk:environment',
+      optionName: 'ServiceRole',
+      value: serviceRole.roleName,
+    },
+    {
+      namespace: 'aws:elasticbeanstalk:environment',
+      optionName: 'LoadBalancerType',
+      value: 'application',
+    },
+    {
+      namespace: 'aws:elasticbeanstalk:managedactions',
+      optionName: 'ServiceRoleForManagedUpdates',
+      value: 'AWSServiceRoleForElasticBeanstalkManagedUpdates',
+    },
+  ],
+});
+
+beanstalkEnv.addDependsOn(instanceProfile);
+beanstalkEnv.addDependsOn(beanstalkApp);
+
+const pipeline = new codepipeline.Pipeline(stack, 'Pipeline', {
+  artifactBucket: bucket,
+});
+
+const sourceOutput = new codepipeline.Artifact('SourceArtifact');
+const sourceAction = new cpactions.S3SourceAction({
+  actionName: 'Source',
+  output: sourceOutput,
+  bucket,
+  bucketKey: Fn.select(0, artifact.objectKeys),
+});
+
+pipeline.addStage({
+  stageName: 'Source',
+  actions: [
+    sourceAction,
+  ],
+});
+
+const deployAction = new cpactions.ElasticBeanstalkDeployAction({
+  actionName: 'Deploy',
+  input: sourceOutput,
+  environmentName: beanstalkEnv.environmentName!,
+  applicationName: beanstalkApp.applicationName!,
+});
+
+pipeline.addStage({
+  stageName: 'Deploy',
+  actions: [
+    deployAction,
+  ],
+});
+
+new integ.IntegTest(app, 'codepipeline-elastic-beanstalk-deploy', {
+  testCases: [stack],
```

Comment:
Can you set `stackUpdateWorkflow: false`. Anytime there is a custom resource in the integ test that
adds a large zip file we should set it to false so that it gets removed. I need to find a way to do
this automatically, but right now it's just in the backlog.

Reply:
Adding!